src/ctx.cpp

```diff
@@ -2995,11 +2995,26 @@ llvm::Value *FunctionEmitContext::CallInst(llvm::Value *func, const FunctionType
 #if ISPC_LLVM_VERSION >= ISPC_LLVM_11_0
         llvm::PointerType *func_ptr_type = llvm::dyn_cast<llvm::PointerType>(func->getType());
         llvm::FunctionType *func_type = llvm::dyn_cast<llvm::FunctionType>(func_ptr_type->getPointerElementType());
-        llvm::Instruction *ci = llvm::CallInst::Create(func_type, func, argVals, name ? name : "", bblock);
+        llvm::CallInst *callinst = llvm::CallInst::Create(func_type, func, argVals, name ? name : "", bblock);
 #else
-        llvm::Instruction *ci = llvm::CallInst::Create(func, argVals, name ? name : "", bblock);
+        llvm::CallInst *callinst = llvm::CallInst::Create(func, argVals, name ? name : "", bblock);
 #endif
 
+        // We could be dealing with a function pointer in which case this will not be a 'llvm::Function'.
+        // If 'llvm::Function', use same calling convention as the actual function definition. It's
+        // important we do this since  prebuilt stdlib functions does not use vectorcall on any OS.
```

Comment:
This sounds like we need to change calling convention for stdlib as well. It should not have bug effect on the resulting code, as most of the functions will be inlined anyway, but I see no reason not to use ``vectorcall`` convetion for stdlib and loose effiency.

Reply:
stdlib.ispc will be compiled with vectorcall. Problem is with the .ll files in builtin.

Reply:
p.s - yes, the code in .ll will be inlined. So, it's not a problem eventually.

Reply:
Left it as it is. Let me know if you disagree.

Reply:
I think it's ok, but if see any non-inlined instances, we need to reconsider this.